disco/disco.go

```diff
@@ -204,7 +214,7 @@ func (d *Disco) discover(hostname svchost.Hostname) (*Host, error) {
 
 	resp, err := client.Do(req)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to request discovery document: %v", err)
```

Comment:
I have a slight concern that we've lost the meaningful text of this error and users would not see exactly which request failed, but rather just the reason ("deadline exceeded waiting for headers", or whatever) and I see a couple of different ways to fix it. The best way may be to modify the `Error` method of `ErrServiceDiscoveryNetworkRequest` above to include it:

`return fmt.Errorf("failed to request discovery document: %w", e.err)`